File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
import heapq

def solution(k, score):
    heap = []
    answer = []
    for sc in score:
        if len(heap) < k:
            heapq.heappush(heap, sc)    
            answer.append(heap[0])
        else:
            heapq.heappush(heap,sc)
            heapq.heappop(heap)
            answer.append(heap[0])
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var rank = [];
    var answer = [];
    for(var sc of score) {
        rank.push(sc);
        rank = rank.sort((a,b) => b-a).slice(0,k);        
        answer.push(rank[rank.length-1]);
    }
 
    return answer;
}